enhancements/installer/openstack-ipi-failure-domains.md

```diff
@@ -0,0 +1,340 @@
+
+Then they'll choose what Failure Domains to use when deploying the control plane and compute nodes by setting the `failureDomain` field in the machine pools:
+
+```yaml
+controlPlane:
+  name: master
+  platform:
+    openstack:
+      failureDomains:
+      - rack1
+      - rack2
+      - rack3
+  replicas: 3 // this will deploy 1 master in each domain
+compute:
+- name: worker
+  platform:
+    openstack:
+      ...
+      failureDomains:
+      - rack1
+      - rack2
+      - rack3
+      - rack4
+      - rack5
+      rootVolume:
+        size: 30
+        type: performance
+  replicas: 10 // this will deploy 2 workers per rack
+```
+
+The installer creates a machineSet per failure domain passed to the compute machinePool.
+
+The machineSet for the compute's `rack1` failure domain will look like the following:
+```yaml
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+    machine.openshift.io/cluster-api-machine-role: worker
+    machine.openshift.io/cluster-api-machine-type: worker
+  name: <infrastructure_ID>-worker-rack1
+  namespace: openshift-machine-api
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+      machine.openshift.io/cluster-api-machineset: <infrastructure_ID>-worker-rack1
+  template:
+    ...
+    spec:
+      metadata:
+      providerSpec:
+        value:
+          apiVersion: openstackproviderconfig.openshift.io/v1alpha1
+          cloudName: openstack
+          flavor: <nova_flavor>
+          kind: OpenstackProviderSpec
+          networks:
+            - subnets:
+              - uuid: d7ffd3d8-87e1-481f-a818-ff2f17787d40
+          availabilityZone: az1
+          rootVolume:
+            diskSize: 30
+            volumeType: performance
+            availabilityZone: cinder-az1
+          ...
+```
+
+The installer will create the control plane machines in a way that they can easily be adopted by the [ControlPlane machineset](https://github.com/openshift/enhancements/blob/master/enhancements/machine-api/control-plane-machine-set.md).
+
+When deploying nodes on separate subnets, the user will need to update the `machineNetwork` option to list all the possible CIDRs on which the machines can be:
+
+```yaml
+networking:
+  ...
+  machineNetwork:
+  - cidr: 192.168.25.0/24
+  - cidr: 192.168.26.0/24
+  - cidr: 192.168.27.0/24
+  - ...
+```
+
+ The `machineNetwork` is the source for creating the security groups rules during installation. There's a [known bug][bz-2095323] in the OpenStack implementation that only takes into account the first CIDR.
+
+For now, `machinesSubnet` is required to be set when deploying nodes on separate Failure Domains, and is then used to identify where to create the API and Ingress ports.
+
+The OpenShift administrator will be able to adjust the number of computes nodes for a specific Failure Domain, but only on day 2.
```

Comment:
I would prefer we omit references to computes.

The OpenShift admin can do this in any case by changing the number of replicas in the machineset they created which they pinned to a particular failure domain.